src/components/GenerationType.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { useEffect, useState } from "react";
+
+export class GenerationType {
+    name: string;
+    amount: number;
+    colour: string; 
+
+    constructor(name : string, amount : number, colour : string) {
+        this.name = name;
+        this.amount = amount;
+        this.colour = colour;
+    }
+}
+
+interface GenerationTypeJSON {
+    fuelType: string;
+    generation: number;
+}
+
+interface GenerationTypeDataPoint {
+    startTime : string;
+    settlementPeriod: number;
+    data: Array<GenerationTypeJSON>;
+}
+
+
+const url = "https://data.dev.elexon.co.uk/bmrs/api/v1/generation/outturn/summary?from=2023-07-12&to=2023-07-12";
+
+const smallestPercentageDisplayed = 0.01;
+
+export interface GenerationTypeChartProps {
+    energyData: GenerationType[];
+}
+
+interface Props {
+    chart: React.FC<GenerationTypeChartProps>;

```

Comment:
So I think there's a simpler way to do this: 
1. Rather than passing a React.FC value as a `prop`, you can set this to `JSX.Element` and call it `children`.
2. Then you can write
```
<GenerationTypeChart>
   <Donut>
</GenerationTypeChart>
```
3. Then you can use this prop in `GenerationTypeChart` via {children}

This has the benefit of:
- not having to export props outside of their classes
- giving a better view of the component hierarchy

Let me know if you have questions regarding this